### ISSUE TITLE
`.degrees()` output type (`tuple` vs. `ETuple`) is inconsistent across polynomial backends 

### ISSUE THREAD

**Issue by maxale** [sagemath/sage#38632]
### Expected Behavior

[Documentation for `.degrees()`](https://doc.sagemath.org/html/en/reference/polynomial_rings/sage/rings/polynomial/multi_polynomial_element.html#sage.rings.polynomial.multi_polynomial_element.MPolynomial_polydict.degrees) says: 
> Returns a tuple (precisely - an `ETuple`) ...

and so the type of its result should be `ETuple`.

### Actual Behavior

In reality, we have
```
sage: R.<x,y,z>=QQ[]
sage: type(x.degrees())
<class 'tuple'>
```

### Additional Information

It is also worth to add an optional parameter `as_ETuples` (similarly to `.exponents(as_ETuples=True)`) that would control the type of function output.

### Environment

```markdown
- **OS**: Ubuntu 24.04.1 LTS
- **Sage Version**: 10.5.beta2
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by pawani27** [comment]
@maxale Can you review my pr for this issue?

**Comment by vneiger** [comment]
There is no bug here. The showed example with `R.<x,y,z>=QQ[]` creates a ring where polynomials will be handled by libsingular. The correct documentation file to look at here is not the given link but [this one](https://doc.sagemath.org/html/en/reference/polynomial_rings/sage/rings/polynomial/multi_polynomial_libsingular.html#sage.rings.polynomial.multi_polynomial_libsingular.MPolynomial_libsingular.degrees). It does not mention `ETuple`. So, as such, the documentation is correct and consistent with the code and behaviour.

However one may indeed consider an enhancement to have consistency between this `degrees()` and the `degrees()` for generic multivariate polynomials (which is correct as well and does return an ETuple). At the same time, one should be careful not to break anything with this change, and maybe some deprecation would be in order (except if the output as an ETuple has the exact same behaviour for basic operations as a python tuple, not sure anymore about this). Any thought on this @maxale ?

**Comment by maxale** [comment]
@vneiger I agree. I have no opinion on whether `ETuple` should be preferred over `tuple` or otherwise, but it's an interface inconsistency that  better be corrected one way or another.
